R/pkgdown.R

```diff
@@ -41,6 +46,52 @@ use_pkgdown <- function(config_file = "_pkgdown.yml", destdir = "docs") {
   invisible(TRUE)
 }
 
+pkgdown_config_path <- function(base_path = proj_get()) {
+  path_first_existing(
+    base_path,
+    c(
+      "_pkgdown.yml",
+      "_pkgdown.yaml",
+      "pkgdown/_pkgdown.yml",
+      "inst/_pkgdown.yml"
+    )
+  )
+}
+
+uses_pkgdown <- function(base_path = proj_get()) {
+  !is.null(pkgdown_config_path(base_path))
+}
+
+pkgdown_config_meta <- function(base_path = proj_get()) {
+  if (!uses_pkgdown(base_path)) {
+    return(list())
+  }
+  path <- pkgdown_config_path(base_path)
+  yaml::read_yaml(path) %||% list()
+}
+
+pkgdown_url <- function(base_path = proj_get(), pedantic = FALSE) {
+  if (!uses_pkgdown(base_path)) {
+    return(NULL)
+  }
+
+  meta <- pkgdown_config_meta(base_path)
+  url <- meta$url
+  if (is.null(url)) {
+    if (pedantic) {
+      ui_warn(
+        "pkgdown config does not specify the site's {ui_field('url')}, \\
+        which is optional but recommended"
+      )
+    }
+    NULL
+  } else {
+    gsub("/$", "", url)
+  }
+}
+
+# travis ----
+
 #' @export
 #' @rdname use_pkgdown
 use_pkgdown_travis <- function() {
```

Comment:
Since you're in here, maybe it's good idea to seriously de-promote `use_pkgdown_travis()` since we no longer recommend it?

Reply:
I already did some of that in one of your PRs before I merged it. Do you think I should do more / different?

https://github.com/r-lib/usethis/pull/1106/commits/90dfa97c75c642c6b31119967f03ff93cef930ef